### ISSUE TITLE
Install Error in Vagrant (new build)

### ISSUE THREAD

**Issue by pbailie** [Submitty/Submitty#4888]
Building a new Vagrant development box encounters an installation error that appears to be a database `NOT NULL` constraint violation regarding electronic gradeables.  Full error log is as follows:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/base.py", line 1246, in _execute_context
    cursor, statement, parameters, context
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/default.py", line 581, in do_execute
    cursor.execute(statement, parameters)
psycopg2.errors.NotNullViolation: null value in column "eg_config_path" violates not-null constraint
DETAIL:  Failing row contains (18_postgres_database, null, f, , 0, f, 1, 2019-12-20 23:59:59-05, t, f, t, f, t, f, 2019-12-14 23:59:59-05, 2019-12-20 23:59:59-05, t, 1, t, 0, 0.5, t, f, 2020-01-01 23:59:59-05, {}, f).


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/bin/setup_sample_courses.py", line 1714, in <module>
    main()
  File "/usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/bin/setup_sample_courses.py", line 208, in main
    courses[course].create()
  File "/usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/bin/setup_sample_courses.py", line 800, in create
    gradeable.create(self.conn, gradeable_table, electronic_table, reg_table, component_table, mark_table)
  File "/usr/local/submitty/GIT_CHECKOUT/Submitty/.setup/bin/setup_sample_courses.py", line 1552, in create
    eg_regrade_request_date=self.regrade_request_date)
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/base.py", line 982, in execute
    return meth(self, multiparams, params)
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/sql/elements.py", line 287, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/base.py", line 1101, in _execute_clauseelement
    distilled_params,
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/base.py", line 1250, in _execute_context
    e, statement, parameters, cursor, context
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/base.py", line 1476, in _handle_dbapi_exception
    util.raise_from_cause(sqlalchemy_exception, exc_info)
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/util/compat.py", line 398, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/util/compat.py", line 152, in reraise
    raise value.with_traceback(tb)
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/base.py", line 1246, in _execute_context
    cursor, statement, parameters, context
  File "/usr/local/lib/python3.6/dist-packages/sqlalchemy/engine/default.py", line 581, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.IntegrityError: (psycopg2.errors.NotNullViolation) null value in column "eg_config_path" violates not-null constraint
DETAIL:  Failing row contains (18_postgres_database, null, f, , 0, f, 1, 2019-12-20 23:59:59-05, t, f, t, f, t, f, 2019-12-14 23:59:59-05, 2019-12-20 23:59:59-05, t, 1, t, 0, 0.5, t, f, 2020-01-01 23:59:59-05, {}, f).

[SQL: INSERT INTO electronic_gradeable (g_id, eg_config_path, eg_is_repository, eg_subdirectory, eg_team_assignment, eg_max_team_size, eg_team_lock_date, eg_use_ta_grading, eg_student_view, eg_student_submit, eg_peer_grading, eg_submission_open_date, eg_submission_due_date, eg_late_days, eg_precision, eg_regrade_request_date) VALUES (%(g_id)s, %(eg_config_path)s, %(eg_is_repository)s, %(eg_subdirectory)s, %(eg_team_assignment)s, %(eg_max_team_size)s, %(eg_team_lock_date)s, %(eg_use_ta_grading)s, %(eg_student_view)s, %(eg_student_submit)s, %(eg_peer_grading)s, %(eg_submission_open_date)s, %(eg_submission_due_date)s, %(eg_late_days)s, %(eg_precision)s, %(eg_regrade_request_date)s)]
[parameters: {'g_id': '18_postgres_database', 'eg_config_path': None, 'eg_is_repository': False, 'eg_subdirectory': '', 'eg_team_assignment': False, 'eg_max_team_size': 1, 'eg_team_lock_date': datetime.datetime(2019, 12, 20, 23, 59, 59, tzinfo=<DstTzInfo 'America/New_York' EST-1 day, 19:00:00 STD>), 'eg_use_ta_grading': True, 'eg_student_view': True, 'eg_student_submit': True, 'eg_peer_grading': False, 'eg_submission_open_date': datetime.datetime(2019, 12, 14, 23, 59, 59, tzinfo=<DstTzInfo 'America/New_York' EST-1 day, 19:00:00 STD>), 'eg_submission_due_date': datetime.datetime(2019, 12, 20, 23, 59, 59, tzinfo=<DstTzInfo 'America/New_York' EST-1 day, 19:00:00 STD>), 'eg_late_days': 1, 'eg_precision': 0.5, 'eg_regrade_request_date': datetime.datetime(2020, 1, 1, 23, 59, 59, tzinfo=<DstTzInfo 'America/New_York' EST-1 day, 19:00:00 STD>)}]
(Background on this error at: http://sqlalche.me/e/gkpj)
```

**Comment by pbailie** [comment]
My testing has shown that the error is caused by a recent addition of the latest tutorial gradeable, "18_postgres_database".

While the configuration appears to be added to Submitty/Tutorial, the repo's release point was not updated.  Therefore, the configuration is not being cloned.  The configuration's absence is interpreted as `NULL`, which violates the course DB's `NOT NULL` constraint on `electronic_gradeable.eg_config_path`.

The fix is to create a new release point for Submitty/Tutorial, and update `.setup/bin/versions.sh` to reflect the new release point.  (may require a repo administrator?)

To build a new Vagrant box in the meantime — in your branch, comment out the properties for "18_postgres_database" (lines 327-344) within `.setup/data/courses/tutorial.yml`.

**Comment by pbailie** [comment]
I tested again.  This issue remains valid.

**Comment by MasterOdin** [comment]
Fixed by #4922
